packages/experiments/src/components/Announced/examples/Announced.Asynchronous.Example.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { Announced } from '../Announced';
+import { createArray, createRef } from 'office-ui-fabric-react/lib/Utilities';
+import { Image } from 'office-ui-fabric-react/lib/Image';
+import { FocusZone } from 'office-ui-fabric-react/lib/FocusZone';
+import { Toggle } from 'office-ui-fabric-react/lib/Toggle';
+import './Announced.Example.scss';
+import { ProgressIndicator } from 'office-ui-fabric-react/lib/ProgressIndicator';
+
+export interface IAnnouncedAsynchronousExampleState {
+  photos: { url: string; width: number; height: number }[];
+  total: number;
+  seconds: number;
+  announced?: JSX.Element;
+  percentComplete: number;
+  loading: boolean;
+}
+
+export interface IAnnouncedAsynchronousExampleProps {}
+
+export class AnnouncedAsynchronousExample extends React.Component<IAnnouncedAsynchronousExampleProps, IAnnouncedAsynchronousExampleState> {
+  private _root = createRef<HTMLElement>();
+  private timer: number;
+  private increaseTotal: number;
+  private delay: number;
+
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      photos: this._createPhotos(),
+      total: 0,
+      seconds: 0,
+      announced: undefined,
+      percentComplete: 0,
+      loading: false
+    };
+
+    this._renderPhotos = this._renderPhotos.bind(this);
+    this._renderAnnounced = this._renderAnnounced.bind(this);
+    this._onFocusPhotoCell = this._onFocusPhotoCell.bind(this);
+    this._onToggleChange = this._onToggleChange.bind(this);
+
+    this.increaseTotal = setInterval(() => {
+      if (this.state.loading && this.state.total < this.state.photos.length) {
+        this.setState({ total: this.state.total + 1 });
+      }
+    }, 2000);
+
+    this.timer = setInterval(() => {
+      if (this.state.loading) {
+        this.setState({ seconds: this.state.seconds + 1 });
+
+        if (this.state.seconds % this.delay === 0) {
+          this.setState({
+            announced: (
+              <Announced message={`${this.state.total}/${this.state.photos.length} photos loaded`} id={'announced-' + this.state.total} />
```

Comment:
Can't the delayed renderer used in the announcer basically have this behavior already?

Reply:
I took another look, it can't but it will only delay the first time the component is mounted.

Reply:
Hmm I'm not entirely sure if it'd do what we want, given that we want to print out the current number of photos loaded so far, and `this.state.total` (total number of photos loaded so far) is changing while the timer is going. So if we passed in `this.state.total` and just specified a delay to pass into the Announced component, I'm not sure that it would handle cases in which `this.state.total` has changed between the time the render is called and the delay. Does that make sense...?

Reply:
I think so, sounds right.